File: basic/src/main/java/com/yuanshenbin/basic/adapter/CommonAdapter.java
package com.yuanshenbin.basic.adapter;


import androidx.annotation.Nullable;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.listener.OnLoadMoreListener;
import com.chad.library.adapter.base.module.LoadMoreModule;

import java.util.List;

/**
 * author : yuanshenbin
 * time   : 2017/9/27
 * desc   :
 */

public abstract class CommonAdapter<T> extends BaseQuickAdapter<T, IBaseViewHolder> implements LoadMoreModule {




    /**
     * @param layoutResId
     * @param data
     */
    public CommonAdapter(int layoutResId, List<T> data) {
        super(layoutResId, data);
        getLoadMoreModule().setOnLoadMoreListener(new OnLoadMoreListener() {
            @Override
            public void onLoadMore() {
                if (l != null) {
                    l.onLoadMore();
                }
            }
        });
        getLoadMoreModule().setEnableLoadMore(false);
    }


    public OnMoreLoadListener l;

    public interface OnMoreLoadListener {
        void onLoadMore();
    }

    public void setOnMoreLoadListener(OnMoreLoadListener l) {
        getLoadMoreModule().setEnableLoadMore(true);
        this.l = l;
    }

}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/adapter/CommonAdapter.kt
package com.yuanshenbin.basic.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.module.LoadMoreModule
import com.chad.library.adapter.base.viewholder.BaseViewHolder

/**
 * author : yuanshenbin
 * time   : 2017/9/27
 * desc   :
 */
abstract class CommonAdapter<T>(layoutResId: Int, data: List<T>?) : BaseQuickAdapter<T, BaseViewHolder>(layoutResId, data as MutableList<T>?), LoadMoreModule {
    var l: OnMoreLoadListener? = null

    interface OnMoreLoadListener {
        fun onLoadMore()
    }

    fun setOnMoreLoadListener(l: OnMoreLoadListener?) {
        loadMoreModule.isEnableLoadMore = true
        this.l = l

    }

    /**
     * @param layoutResId
     * @param data
     */
    init {
        loadMoreModule.setOnLoadMoreListener {
            l?.onLoadMore()
        }
        loadMoreModule.isEnableLoadMore = false
    }
}